### PR TITLE
Feature/234 comunication auth

### PR DESCRIPTION
**Pull request by Rebal67**

### CODE REVIEW

File: src/keymanager.ts [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
+    KeyManagerLevel,
+    sha256,
+    SignDataOptions,
+    StoreKeyOptions,
+} from '@tonomy/tonomy-id-sdk';
 
 export default class JsKeyManager implements KeyManager {
-    signData(options: SignDataOptions): Promise<string | Signature> {
-        throw new Error('Method not implemented.');
+    async signData(options: SignDataOptions): Promise<string | Signature> {
```
Comment (theblockstalk) [inline src/keymanager.ts:13]: I'm thinking that the SDK should be providing all these... both the JsKeyManager and the browser key mangares. New Task

File: src/pages/CallBack.tsx [hunk 2]
```diff
@@ -13,8 +13,8 @@ export default function CallBackPage() {
         const { result, accountName, username } = await UserApps.onAppRedirectVerifyRequests();
 
         console.log(result, accountName);
-        const redirectJwt = result.find((jwtVerified) => jwtVerified.payload.origin !== location.origin);
-        const ssoJwt = result.find((jwtVerified) => jwtVerified.payload.origin === location.origin);
+        const redirectJwt = result.find((jwtVerified) => jwtVerified.getPayload().origin !== location.origin);
```
Comment (theblockstalk) [inline src/pages/CallBack.tsx:16]: requests should be a VC, not a message. New Task

File: src/pages/Login.tsx [hunk 3]
```diff
@@ -38,18 +37,45 @@ function Login() {
...
-
-            communication.SSOWebsiteSendToMobile(randomSeed, requests);
-            setShowQR(randomSeed);
-
-            communication.onJwtToClient((data) => {
-                console.log(data);
-                window.location.replace(
-                    `/callback?requests=${data.requests}&accountName=${data.accountName}&username=nousername`
-                );
+            const logInMessage = new Message(jwtRequests[1]);
```
Comment (theblockstalk) [inline src/pages/Login.tsx:41]: requests should be VC, not Message. New Task

File: src/pages/Login.tsx [hunk 4]
```diff
@@ -38,18 +37,45 @@ function Login() {
...
+            setShowQR(did);
+
+            await communication.login(logInMessage);
+
+            communication.subscribeMessage(async (responseMessage) => {
+                const message = new Message(responseMessage);
+
+                console.log('recieved', message);
+
+                if (message.getPayload().type === 'ack') {
```
Comment (theblockstalk) [inline src/pages/Login.tsx:61]: please add some comments here describing the flow

File: src/pages/CallBack.tsx [hunk 5]
```diff
@@ -33,7 +33,7 @@ export default function CallBackPage() {
             }
         }
 
-        const redirectJwtPayload = redirectJwt.payload;
+        const redirectJwtPayload = redirectJwt.getPayload();
         const url =
```
Comment (theblockstalk) [inline src/pages/CallBack.tsx:37]: follow up is we need to set storage to have the account name and DID of the account, which should also be checked when the user opens the site to see if they are already logged in. New Task

Comment (Rebal67) [reply src/pages/CallBack.tsx:37]: agreed

File: src/pages/Login.tsx (hunk 3 above)
Comment (Rebal67) [reply src/pages/Login.tsx:41]: agreed

File: src/pages/CallBack.tsx (hunk 2 above)
Comment (Rebal67) [reply src/pages/CallBack.tsx:16]: agreed

File: src/keymanager.ts (hunk 1 above)
Comment (Rebal67) [reply src/keymanager.ts:13]: task is already written #396

**Pull request merged**